### PR TITLE
feat(docs): Adding example of managing the flux ssh keypair in terraform.

### LINKED ISSUE #646: [Repo]: Add a Terraform example for self managing SSH keypair
### Description

This example would show how you can manage the SSH keypair that flux uses in Terraform to allow for easier rotation.

### References

Was discussed in https://github.com/fluxcd/terraform-provider-flux/issues/596

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by swade1987**
## Description
Adds an example showing how to manage the SSH keypair flux uses in Terraform for easier rotation.

## Motivation and Context
Fixes #646 
Discussed in https://github.com/fluxcd/terraform-provider-flux/issues/596

## How has this been tested?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation
- [x] I have updated the documentation (if required) with `make docs`

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/fluxcd/terraform-provider-flux/blob/main/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

## Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ All documentation is available on the [Terraform provider website](https://regis
 
 The following guides are available to help you use the provider:
 
+- [Bootstrapping a cluster using a GitHub repository self managing SSH keypair secret)](examples/github-self-managed-ssh-keypair)
```
Comment (stefanprodan) [inline README.md:20]: Move this below after GPG

File: examples/github-self-managed-ssh-keypair/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# GitHub via SSH
+
+The example demonstrates how to bootstrap a KinD cluster with Flux using a GitHub repository via SSH.
+
+However, this time the SSH keypair is managed by Terraform. So it can be rotated as needed.
```
Comment (stefanprodan) [inline examples/github-self-managed-ssh-keypair/README.md:5]:
```suggestion
The SSH keypair is managed by Terraform and can be rotated as needed.
```

File: templates/index.md.tmpl [hunk 3]
```diff
@@ -15,9 +15,11 @@ The provider needs to be configured with Kubernetes credentials to be used.
 
 The following examples are available to help you use the provider:
 
+- [Bootstrapping a cluster using a GitHub repository self managing SSH keypair secret)](https://github.com/fluxcd/terraform-provider-flux/tree/main/examples/github-self-managed-ssh-keypair)
```
Comment (stefanprodan) [inline templates/index.md.tmpl:18]: Remove this one

Comment (swade1987) [reply templates/index.md.tmpl:18]: This has to stay, its used to render documentation when running `make docs`

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @swade1987

**Pull request merged**